java/drivers/driver-hazelcast4plus/src/main/java/com/hazelcast/simulator/tests/map/prunability/ScanByPrunedCompositeKeyBenchmarkRandomAccess.java

```diff
@@ -0,0 +1,116 @@
+
+import com.hazelcast.config.MapConfig;
+import com.hazelcast.config.PartitioningAttributeConfig;
+import com.hazelcast.map.IMap;
+import com.hazelcast.simulator.hz.HazelcastTest;
+import com.hazelcast.simulator.hz.IdentifiedDataSerializablePojo;
+import com.hazelcast.simulator.test.annotations.Prepare;
+import com.hazelcast.simulator.test.annotations.Setup;
+import com.hazelcast.simulator.test.annotations.Teardown;
+import com.hazelcast.simulator.test.annotations.TimeStep;
+import com.hazelcast.simulator.worker.loadsupport.Streamer;
+import com.hazelcast.simulator.worker.loadsupport.StreamerFactory;
+import com.hazelcast.sql.SqlResult;
+import com.hazelcast.sql.SqlRow;
+import com.hazelcast.sql.SqlService;
+
+import java.io.Serializable;
+import java.util.List;
+import java.util.Random;
+
+import static java.util.Arrays.asList;
+
+public class ScanByPrunedCompositeKeyBenchmarkRandomAccess extends HazelcastTest {
+
+    // properties
+    // the number of map entries
+    public int entryCount = 500_000;
+
+    private IMap<KeyPojo, String> map;
+    private SqlService sqlService;
+    private String query;
+
+    @Setup
+    public void setUp() {
+        final List<PartitioningAttributeConfig> attributeConfigs = asList(
+                new PartitioningAttributeConfig("a"),
+                new PartitioningAttributeConfig("c")
+        );
+        final MapConfig mapConfig = new MapConfig(name).setPartitioningAttributeConfigs(attributeConfigs);
+        targetInstance.getConfig().addMapConfig(mapConfig);
+        this.sqlService = targetInstance.getSql();
+        this.map = targetInstance.getMap(name);
+
+        this.query = "SELECT this FROM " + name + " WHERE a = ? AND c = ?";
+    }
+
+    @Prepare(global = true)
+    public void prepare() {
+        Streamer<KeyPojo, String> streamer = StreamerFactory.getInstance(map);
+
+        for (int i = 0; i < entryCount; i++) {
+            Integer iKey = i;
+            Long lKey = (long) i;
+            String v = String.format("%010d", iKey);
+            KeyPojo keyPojo = new KeyPojo(iKey, v, lKey);
+            streamer.pushEntry(keyPojo, v);
+        }
+        streamer.await();
+
+        SqlService sqlService = targetInstance.getSql();
+        String createMappingQuery = "CREATE EXTERNAL MAPPING IF NOT EXISTS " + name + " "
+                + "EXTERNAL NAME " + name + " "
+                + "        TYPE IMap\n"
+                + "        OPTIONS (\n"
+                + "                'keyFormat' = 'java',\n"
+                + "                'keyJavaClass' = 'com.hazelcast.simulator.tests.map.prunability.KeyPojo',\n"
+                + "                'valueFormat' = 'java',\n"
+                + "                'valueJavaClass' = 'java.lang.String'\n"
+                + "        )";
+
+        sqlService.execute(createMappingQuery);
+    }
+
+    @TimeStep
+    public void timeStep() throws Exception {
+        int actual = 0;
+        int key = new Random().nextInt(entryCount);
```

Comment:
this line is the only difference between both tests, everything else is copypasted. It would be easier to understand that we test the same if the common code was in single place.